mcf-browser/src/back-end/parse-tmcf.ts

```diff
@@ -174,6 +175,114 @@ class ParseTmcf {
     return mcfLines.join('\n');
   }
 
+  /**
```

Comment:
feels like these parsing logic shouldn't be repeated and doesn't belong in parse-tmcf?

Reply:
I agree, I'll refactor this code for the performance optimization PR if that's okay (see comment above)

Reply:
ok